datafusion/expr/src/udaf.rs

```diff
@@ -153,9 +153,12 @@ impl AggregateUDF {
     }
 
     /// Return an accumulator the given aggregate, given its return datatype
-    pub fn accumulator(&self, return_type: &DataType) -> Result<Box<dyn Accumulator>> {
-        let sort_exprs = self.inner.sort_exprs();
-        let schema = self.inner.schema();
+    pub fn accumulator(
```

Comment:
This seems like it is the API that would make sense, but I am still not 100% sure -- the Accumulator simply gets `ArrayRef` (not `RecordBatch`) so it currently doesn't know anything about the schema 🤔 

Maybe we need to study how the existing built in aggregate functions that do support order by handle representing the ordering and follow the same model in UDAF